### ISSUE TITLE
Problems with Salted Extension and Error Number #1533818591

### ISSUE THREAD

**Issue by sarah12typo3** [in2code-de/femanager#146]
Using: Typo3 9.5.4, femanager 5.0.0, build-in felogin

I am trying to create a user registration with the femanager. The creation of users with username, email and password works fine. But once I want to login - using a fomular from felogin, I get an Typo3 Error, and the log reports that this is an "Uncaught TYPO3 Exception: #1533818591".
But the wiki entry for #1533818591 doesn't help to fix the problem. 

Looking into the code, we observed that when commenting out one line in "typo3conf\ext\femanager\Classes\Utility\UserUtility.php" --> line 185 and 190 it works.

```
185: // if (ExtensionManagementUtility::isLoaded('saltedpasswords')) {
190: // }
```
We figured, that this is due to the fact, that the extension is merged to the core, therefore an if-clause is not necessary, indeed it throws an error. 
see "Merge salted passwords extension into core": https://forge.typo3.org/issues/85833

**Comment by McKoy61** [comment]
Same problem, same constellation (Typo3 9.5.4, femanager 5.0.0, build-in felogin
Neither the solutions mentioned in the wiki entry of the uncaught TYPO3 Exception #1533818591, nore uncomenting as described above did help. Only resetting the password with the forgotten-password-link from felogin makes the login-procedure work, which is not a useful workaround.

**Comment by wtfred** [comment]
Just bump into this problem, on TYPO3 9.5
In class `femanager/Classes/Utility/UserUtility.php`, in the `hashPassword` method, the `default` part at line 185 should now be:
```
default:
    $hashInstance = GeneralUtility::makeInstance(PasswordHashFactory::class)->getDefaultHashInstance('FE');
    $user->setPassword($hashInstance->getHashedPassword($user->getPassword()));
```

**Comment by McKoy61** [comment]
> Just bump into this problem, on TYPO3 9.5
> In my opinion, in class `femanager/Classes/Utility/UserUtility.php`, in the `hashPassword` method, the `default` part at line 185 should now be:
> 
> ```
> default:
>     $hashInstance = GeneralUtility::makeInstance(PasswordHashFactory::class)->getDefaultHashInstance('FE');
>     $user->setPassword($hashInstance->getHashedPassword($user->getPassword()));
> ```
> 
> (i'm using develop branch as 5.0.0 looks not compatible)

@wtfred Thanks for your suggestion, but this throws another error on registration: 
Class 'In2code\Femanager\Utility\PasswordHashFactory' not found

**Comment by TheNaderio** [comment]
@McKoy61 you have to add either the use-statement at the top of the class:
`use TYPO3\CMS\Core\Crypto\PasswordHashing\PasswordHashFactory;`

or the full qualified name of the class:
```
$hashInstance = GeneralUtility::makeInstance(\TYPO3\CMS\Core\Crypto\PasswordHashing\PasswordHashFactory::class)->getDefaultHashInstance('FE');
$user->setPassword($hashInstance->getHashedPassword($user->getPassword()));
```

**Comment by McKoy61** [comment]
> @McKoy61 you have to add either the use-statement at the top of the class:
> `use TYPO3\CMS\Core\Crypto\PasswordHashing\PasswordHashFactory;`
> 
> or the full qualified name of the class:
> 
> ```
> $hashInstance = GeneralUtility::makeInstance(\TYPO3\CMS\Core\Crypto\PasswordHashing\PasswordHashFactory::class)->getDefaultHashInstance('FE');
> $user->setPassword($hashInstance->getHashedPassword($user->getPassword()));
> ```
Thanks for the hint, this error disappeared now, but the problem itself is not solved.

**Comment by TheNaderio** [comment]
@McKoy61 Please register a new user and try again with the new one.

**Comment by McKoy61** [comment]
> @McKoy61 Please register a new user and try again with the new one.
Thats how I did it, same problem.

**Comment by wtfred** [comment]
Maybe try this on the develop branch, that's the one i used, did the trick for me.

**Comment by PeterMeiningen** [comment]
Same Problem Typo9.5.4 Femanager 5.0.0 
After create and felogin error 153...……… Validation Serverside enabled, clientside 0. 
Argon2i not available Passwort hashing bcrypt. Need Help

**Comment by McKoy61** [comment]
> Maybe try this on the develop branch, that's the one i used, did the trick for me.

Same on develop branch.

**Comment by Footrotflat** [comment]
I made a PR #148 for hashing with argon2i as default.

**Comment by PeterMeiningen** [comment]
The change of footrotflat works for me, Thank you

**Comment by McKoy61** [comment]
> I made a PR #148 for hashing with argon2i as default.

Thanks! Spent some time in testing. argon2i ist available and configured, php 7.2 is in use. Same exception thrown on trying to log in with the new password registered in femanager.  but the passwords are stored completely unhashed in the database (clear text). Resetting the password with the password forgotten-link of felogin, the password ist stored as argon2i-hash, log-in is successful. Meanwhile femanager was completely uninstalled and deleted, database analyzed and suggested changes made, all cashes deleted, and made a new install (delvelop-branche). Behaviour does not change. Finally switched to php7.3 just for fun, the exception disappears, but passwords are furthermore not hashed.

**Comment by Footrotflat** [comment]
@McKoy61 and don't forget to set

Settings => Configuration Presets => set Argon2i as default

**Comment by McKoy61** [comment]
> @McKoy61 and don't forget to set
> 
> Settings => Configuration Presets => set Argon2i as default

Yes, Argon2i was already set in Configure Settings/Installation-Wide Options/[BE][passwordHashing][className] and in Settings/Configuration Presets/Password Hashing Settings.

**Comment by sbusemann** [comment]
this is solved in version 5.1.0
